src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InitializeHandler.cs

```diff
@@ -47,6 +48,10 @@ internal class InitializeHandler : IRequestHandler<InitializeParams, InitializeR
                 },
                 ImplementationProvider = true,
                 SupportsDiagnosticRequests = true,
+                InlineCompletionOptions = new VSInternalInlineCompletionOptions
+                {
+                    Pattern = new Regex(string.Join("|", InlineCompletionHandler.CSharpKeywords))
```

Comment:
this is rather hardcoded, but the plan is to eventually not need the pattern once the client side is fully async.  So imho not worth doing anything super complicated here.

Reply:
Totally agree!

Even though it literallllly means nothing in practice lets also add `, RegexOptions.Compiled | RegexOptions.Singleline, TimeSpan.FromSeconds(1))` to the regex just for any future on-lookers it's easy to see the pattern we typically follow for Regex.

Reply:
I am all for anything and everything we can do to get rid of Regex. Especially in a spot like this where the usage is super simple. Is there an issue filed about the ClientSide async stuff, and does it specifically call out the pattern for removal? If so I'm happy (after Taylors Regex safety changes above).

Reply:
Actually, on second thought is this pattern really correct? Wouldn't this match things like `therefor` or `format` (which I assume it should not)? Should we add `\b` to the front/and/or/end of each keyword, or does the platform already do that for us?

Reply:
Would this trigger on:
```razor
@code{
    var str = "foreach";
}
```
? Or is that fine because we'll just let the C# LS return no results?

Reply:
>Is there an issue filed about the ClientSide async stuff, and does it specifically call out the pattern for removal? If so I'm happy (after Taylors Regex safety changes above).

Finding it... will link here once I do

>Actually, on second thought is this pattern really correct? Wouldn't this match things like therefor or format (which I assume it should not)? Should we add \b to the front/and/or/end of each keyword, or does the platform already do that for us?

It would, I can add \b, but note 

> Would this trigger on:

Yes, the regex is just an optimization due to the blocking nature of the request.  It is up to the language server to make the final decision as to whether or not there is actually a snippet there.